tests/ocp/sriov/internal/sriovenv/sriovenv.go

```diff
@@ -0,0 +1,2402 @@
+	}
+
+	err = policyBuilder.Delete()
+	if err != nil {
+		return fmt.Errorf("failed to delete SRIOV policy %q: %w", name, err)
+	}
+
+	// Wait for policy to be deleted using wait.PollUntilContextTimeout
+	klog.V(90).Infof("Waiting for SRIOV policy %q to be deleted", name)
+
+	err = wait.PollUntilContextTimeout(
+		context.Background(),
+		tsparams.PollingInterval,
+		timeout,
+		true,
+		func(ctx context.Context) (bool, error) {
+			_, pullErr := sriov.PullPolicy(apiClient, name, sriovOpNs)
+
+			return isPolicyNotFoundError(pullErr), nil
+		})
+	if err != nil {
+		return fmt.Errorf("timeout waiting for SRIOV policy %q to be deleted from namespace %q: %w",
+			name, sriovOpNs, err)
+	}
+
+	klog.V(90).Infof("SRIOV policy %q successfully deleted", name)
+
+	return nil
+}
+
+// ErrNetworkNotFound is returned when a SRIOV network is not found.
+var ErrNetworkNotFound = fmt.Errorf("sriov network not found")
+
+// findSriovNetwork finds a SRIOV network by name and returns its target namespace.
+// Returns ErrNetworkNotFound if the network does not exist.
+func findSriovNetwork(
+	apiClient *clients.Settings,
+	name, sriovOpNs string,
+) (*sriov.NetworkBuilder, string, error) {
+	networks, err := sriov.List(apiClient, sriovOpNs, client.ListOptions{})
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to list networks: %w", err)
+	}
+
+	for _, network := range networks {
+		if network.Object.Name == name {
+			targetNs := network.Object.Spec.NetworkNamespace
+			if targetNs == "" {
+				targetNs = sriovOpNs
+			}
+
+			builder := sriov.NewNetworkBuilder(
+				apiClient, name, sriovOpNs, targetNs, network.Object.Spec.ResourceName)
+
+			return builder, targetNs, nil
+		}
+	}
+
+	return nil, "", fmt.Errorf("%w: %q in namespace %q", ErrNetworkNotFound, name, sriovOpNs)
+}
+
+// isNADNotFoundError checks if the error indicates the NAD doesn't exist.
+// This handles both Kubernetes API NotFound errors and eco-goinfra custom errors.
+func isNADNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if apierrors.IsNotFound(err) {
+		return true
+	}
+
+	// eco-goinfra returns custom error messages like "does not exist"
+	errMsg := err.Error()
+
+	return strings.Contains(errMsg, "does not exist") ||
+		strings.Contains(errMsg, "not found")
+}
+
+// isNetworkNotFoundError checks if the error indicates the SriovNetwork doesn't exist.
+// This handles both Kubernetes API NotFound errors and eco-goinfra custom errors.
+func isNetworkNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	if apierrors.IsNotFound(err) {
+		return true
+	}
+
+	// eco-goinfra returns custom error messages like "does not exist"
+	errMsg := err.Error()
+
+	return strings.Contains(errMsg, "does not exist") ||
+		strings.Contains(errMsg, "not found")
+}
+
+// waitForNADDeletion waits for a NAD to be deleted.
+func waitForNADDeletion(apiClient *clients.Settings, name, namespace string) error {
```

Comment:
We have the similar func. Please move it higher level fro reuse if it's ok
https://github.com/rh-ecosystem-edge/eco-gotests/blob/main/tests/cnf/core/network/sriov/internal/sriovenv/sriovenv.go#L110